### PR TITLE
Deserializer rounding

### PR DESCRIPTION
**Pull request by jaeyoo**
This PR fixes precision issue during serialization & deserialization.

### CODE REVIEW

**Comment by jaeyoo** [comment]
> Feel free to merge. Can we up this to 7 or 8 or does it become unstable at that point ?

No we can't. It shows discrepancy at 7 or 8. Thanks.

File: tensorflow_quantum/core/serialize/serializer.py [hunk 1]
```diff
@@ -26,6 +27,10 @@
 _CONSTANT_TRUE = lambda x: True
 
 
+def _round(x):
+    return np.round(x, 6) if isinstance(x, float) else x
```
Comment (MichaelBroughton) [inline tensorflow_quantum/core/serialize/serializer.py:31]: Are we ok with the fact that:
```python3
isinstance(np.float32(0.1), float) => False
```
 ?

Comment (jaeyoo) [reply tensorflow_quantum/core/serialize/serializer.py:31]: hmm we don't use `np.float32`. if we use, even if the value itself is exactly the same, circuit equality failed due to types.

Comment (jaeyoo) [reply tensorflow_quantum/core/serialize/serializer.py:31]: so it is intended.

Comment (jaeyoo) [reply tensorflow_quantum/core/serialize/serializer.py:31]: The reason why I failed to update serializer was using `np.float32`. I realized that `np.round` itself doesn't require the data type, so I could do it today :)

**Pull request merged**